ARCHITECTURE.md

```diff
@@ -0,0 +1,197 @@
+- Optimize the critical path that reduces your SDLC's cycle time.
+
+Additionally, unlike similar projects, we harness the power of Nix & Flakes to ensure reproducibility.
+
+## Goals
+
+- _Complete_: Standard should cover the important use cases for setting up and running the automatable sections of the SDLC.
+- _Optimized_: Standard should optimize both for the needs of the individual developers and the market success of the product.
+- _Integrated_: Standard should provide the user with a satisfying integration experience across a well-curated assortment of tools and functionality.
+
+Please defer to the [sales pitch](./PITCH.md), if you need more excitement.
+
+## Ideals
+
+While we aim to improve the SDLC by applying Nix and its ecoysystem's ingenuity to the problem, we also want to build bridges.
+In order to bring the powers of store based reproducible packaging to colleagues and friends, we need to maneuver around the ecosystem's pitfalls:
+
+- _Use nix only where it is best suited_ &mdash; a Nix maximalist approach may be an innate condition to some of us, but to build bridges we deeply recognize and value other perspectives and don't dismiss them as ignorance.
+- _Disrupt where disruption is necessary_ &mdash; the Nix ecosystem has a fairly rigid set of principles and norms that we don't think always apply in our use cases.
+- _Look left, right, above and beyond_ &mdash; our end-to-end perspective commands us to actively seek and reach out to other projects and ecosystems to compose our value chain; there's no place for the "not invented here"-syndrome.
+
+## Scope
+
+These are big goals and ideals.
+In the interest of practical advancements, we'll narrow down the scope in this section.
+
+We can subdivide (not break up!) our process into roughly three regions with different shapes and characteristics:
+
+- **Development Environment** roughly covers _code-to-commit_.
+- **Packaging Pipeline** roughly covers _commit-to-distribution_.
+- **Deployment and Beyond** roughly covers _distribution-to-next-rollout_.
+
+We delegate:
+
+- The **Development Environment** to a trusted project in the broader Nix Community employing community outreach to promote our cause and ensure it is at least not accidentally sabotaged.
+- The **Deployment and Beyond** by cultivating outreach and dovetailing with initiatives of, among others, the Cloud Native ecosystem.
+
+And we focus on:
+
+- The **Packaging Pipeline**
+- _Interfaces_ and _Integration_ with the other two
+
+## Architecture
+
+### Overview
+
+#### Locating Standard in the SDLC
+
+Where is Standard located in the big picture?
+
+This graphic locates Standard across the SDLC & ALM.
+
+But not only that.
+It also explains how automation in itself is implemented as _code_, just as the application itself.
+Therefore, we make a distinction between:
+
+- first order application code on the first level (L1); and
+- higher order supporting code on higher levels as exemplified by L2 and L3.
```

Comment:
maybe reword as:
"above that, higher order supporting code as exemplified by L2 and L3."